rubix-bookkeeper/src/test/java/com/qubole/rubix/bookkeeper/TestBookKeeper.java

```diff
@@ -100,6 +101,43 @@ public void testGetDummyClusterManagerValidInstance() throws Exception
         " Expected : " + DummyClusterManager.class + " Got : " + manager.getClass());
   }
 
+  @Test
+  private void testGetCacheDirSizeinMBs()
+          throws IOException, TException
+  {
+    String testDirectory = CacheConfig.getCacheDirPrefixList(conf) + "0" + CacheConfig.getCacheDataDirSuffix(conf);
+    CacheConfig.setBlockSize(conf, 1 * 1024 * 1024);
+    String backendFileName = testDirectory + "testBackendFile";
+    DataGen.populateFile(backendFileName, 1, 10000000);
+    final String remotePathWithScheme = "file://" + backendFileName;
+    bookKeeper.readData(remotePathWithScheme, (long) 100000000, 2500000, 102500000, TEST_LAST_MODIFIED, ClusterType.TEST_CLUSTER_MANAGER.ordinal());
+
+    // read from randomAccessFile and verify that it has the right data
+    // 0 - 99614719 : should be filled with '0'
+    // 99614720 - 99999999 : should be filled with 'j'
+    // 100000000 - 102500000 : should be filled with 'k'
+    byte[] buffer = new byte[260000000];
+    FileInputStream localFileInputStream = new FileInputStream(new File(testDirectory + backendFileName));
```

Comment:
This is reading from the source file. When we are verifying data, we should be reading from downloaded data not from the source data.

Reply:
It is reading from downloaded file only not from source file. The variable names may be misleading. To avoid confusion, used test util function to get downloaded file.